nox/virtualenv.py

```diff
@@ -174,6 +175,26 @@ def create(self):
         return True
 
 
+def resolve_real_python_outside_venv(desired_intepreter: str) -> str:
+    """Return path to the real Python installation based
+
+    See also:
+    https://docs.python.org/3/library/sys.html#sys.prefix
+    https://docs.python.org/3/library/sys.html#sys.base_prefix
+    """
```

Comment:
This definitely won't work on Windows. There are various implementations of this type of "locate Python" algorithm on the web, including a reusable library version at https://github.com/sarugaku/pythonfinder.

Unfortunately, reusable code for this seems hard to locate (hence the reason so many people write their own) but IMO we should use a library if at all possible (pythonfinder is the one I've heard of, but I have no problem if someone recommends a better one).

Reply:
pythonfinder looks like it can replace the entire existing nox function unless I'm missing something. Is that correct?

Reply:
I replaced all the code with pythonfinder but it isn't able to find any installations. I'll keep trying with it. If it works it will simplify the code tremendously.

Reply:
I've not used pythonfinder myself, I just knew about it from other projects. So I've no experience with it I'm afraid. I'll try to have a play with it myself, in case I get any ideas.